skimage/draw/_draw.pyx

```diff
@@ -347,18 +342,14 @@ def circle_perimeter(Py_ssize_t cy, Py_ssize_t cx, Py_ssize_t radius,
            [0, 0, 0, 1, 1, 1, 0, 0, 0, 0],
            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=uint8)
-
     """
 
     cdef list rr = list()
     cdef list cc = list()
 
     cdef Py_ssize_t x = 0
-    cdef Py_ssize_t y = radius
-    cdef Py_ssize_t d = 0
-
-    cdef double dceil = 0
-    cdef double dceil_prev = 0
+    cdef Py_ssize_t y = <int> round(radius)
```

Comment:
I don't know if it's not correct, but for sure, not ideal :)

For Bresenham, we must consider the closest pixels from the circle. For Andres, it might be more subtle.
And the same work must be done for other shapes for consistency.